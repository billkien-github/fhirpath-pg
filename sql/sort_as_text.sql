\set pt $JSON${"resourceType":"Patient", "name": [{"use": "official", "ext": {"bext" : ["fext"]}, "family": ["ivanov", "vodkin"], "given": ["petrov"]}, {"use": "common", "family": ["petrov"]}]}$JSON$

SELECT fhirpath_sort_as_text(:pt, '.name', 'HumanName');
SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "family": ["f1", "f2"]}]}', '.name', 'HumanName');
SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "line": ["f1", "f2"]}]}', '.name', 'Address');
SELECT fhirpath_sort_as_text(:pt, '.name.family', 'string');
SELECT fhirpath_sort_as_text(:pt, '.name.where(use=official).family', 'HumanName');
SELECT fhirpath_sort_as_text(:pt, '.name.where(use=common).family', 'HumanName');
SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": {"start":  "SKIP", "end": "SKIP"}, "line": ["f1", "f2"]}]}', '.name', 'Address');
SELECT fhirpath_sort_as_text('{"a": 222}', '.a', 'Numeric');
SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "given": ["Tanebaum"], "family": ["Rudolf"]}]}', '.name', 'HumanName');
SELECT fhirpath_sort_as_text('{"coding": {"code": "C1", "system": "SYS"}}', '.coding', 'Coding');
