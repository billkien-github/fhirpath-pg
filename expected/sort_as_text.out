\set pt $JSON${"resourceType":"Patient", "name": [{"use": "official", "ext": {"bext" : ["fext"]}, "family": ["ivanov", "vodkin"], "given": ["petrov"]}, {"use": "common", "family": ["petrov"]}]}$JSON$
SELECT fhirpath_sort_as_text(:pt, '.name', 'HumanName');
     fhirpath_sort_as_text      
--------------------------------
 ["ivanov", "vodkin"]["petrov"]
(1 row)

SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "family": ["f1", "f2"]}]}', '.name', 'HumanName');
 fhirpath_sort_as_text 
-----------------------
 ["f1", "f2"]
(1 row)

SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "line": ["f1", "f2"]}]}', '.name', 'Address');
 fhirpath_sort_as_text 
-----------------------
 ["f1", "f2"]
(1 row)

SELECT fhirpath_sort_as_text(:pt, '.name.family', 'string');
 fhirpath_sort_as_text 
-----------------------
 ivanov
(1 row)

SELECT fhirpath_sort_as_text(:pt, '.name.where(use=official).family', 'HumanName');
 fhirpath_sort_as_text 
-----------------------
 ivanov
(1 row)

SELECT fhirpath_sort_as_text(:pt, '.name.where(use=common).family', 'HumanName');
 fhirpath_sort_as_text 
-----------------------
 petrov
(1 row)

SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": {"start":  "SKIP", "end": "SKIP"}, "line": ["f1", "f2"]}]}', '.name', 'Address');
 fhirpath_sort_as_text 
-----------------------
 ["f1", "f2"]
(1 row)

SELECT fhirpath_sort_as_text('{"a": 222}', '.a', 'Numeric');
 fhirpath_sort_as_text 
-----------------------
 222
(1 row)

SELECT fhirpath_sort_as_text('{"name": [{"use": "SKIP", "period": "SKIP", "given": ["Tanebaum"], "family": ["Rudolf"]}]}', '.name', 'HumanName');
 fhirpath_sort_as_text  
------------------------
 ["Rudolf"]["Tanebaum"]
(1 row)

SELECT fhirpath_sort_as_text('{"coding": {"code": "C1", "system": "SYS"}}', '.coding', 'Coding');
 fhirpath_sort_as_text 
-----------------------
 SYSC1
(1 row)

SELECT fhirpath_sort_as_text('{"ref": {"reference": "pt/1"}}', '.ref', 'Reference');
 fhirpath_sort_as_text 
-----------------------
 pt/1
(1 row)

